src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1217,8 +1219,15 @@ void MergeTreeData::clearPartsFromFilesystem(const DataPartsVector & parts_to_re
         /// NOTE: Under heavy system load you may get "Cannot schedule a task" from ThreadPool.
         for (const DataPartPtr & part : parts_to_remove)
         {
-            pool.scheduleOrThrowOnError([&]
+            pool.scheduleOrThrowOnError([&, thread_group = CurrentThread::getGroup()]
```

Comment:
@azat I did not investigrate this code a lot before, can you explain why we need this code with thread_group ?

Reply:
Otherwise those threads will not be accounted for the query (i.e. `drop table` query in `system.query_log` will not include used threads in `thread_ids` column), this done as a separate patch.

Reply:
Make sense.